### PR TITLE
Add investing query parameters and +37 currencies support

### PR DESCRIPTION
**Pull request by gustavomfc**
Add possibility to custom investing query results, with time interval between results, number of results and period of the results.
Add suport to following currency codes:

-USD/CHF
-EUR/GBP
-USD/CAD
-GBP/JPY
-GBP/CHF
-CHF/JPY
-CAD/CHF
-EUR/AUD
-EUR/CAD
-USD/ZAR
-USD/TRY
-ARS/BRL
-AUD/BRL
-BRL/ARS
-BRL/AUD
-BRL/CAD
-BRL/CHF
-BRL/CLP
-BRL/DKK
-BRL/HKD
-BRL/JPY
-BRL/MXN
-BRL/SGD
-BRL/USD
-CAD/BRL
-CHF/BRL
-EUR/BRL
-GBP/BRL
-JPY/BRL
-SGD/BRL
-USD/BRL
-AED/BRL
-CNY/BRL
-DKK/BRL
-HKD/BRL
-NOK/BRL
-SEK/BRL

### CODE REVIEW

**Comment by DavideViolante** [comment]
Hi, thanks for the contribution. Before merging:

1. Fix the linting errors, see above checks
2. Add the 3 new params as optional parameters to prevent breaking changes and being able to not specify them. You can do it [like that](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters). The API should work with 1 param too.
   - Example 1: `await investing('currencies/eur-usd')`
   - Example 2 `await investing('currencies/eur-usd', 3600, 24, '1-day')`
   - So the function investing becomes `async function investing (input, interval = 3600, candle_count = 24, period = '1-day') {`
3. Sort mapping.js by pairId please
4. Add new tests to test the API with the new optional params, test edge cases too

**Comment by gustavomfc** [comment]
Hi! Thanks for your review and comments
I've just made all the suggested changes

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ const { investing } = require('investing-com-api')
 
 async function main () {
   try {
-    const response = await investing('currencies/eur-usd')
+    const response = await investing('currencies/eur-usd', 3600, 24, '1-day') //EURxUSD quotes of last 24h (input, interval, candle_count, period)
```
Comment (DavideViolante) [inline README.md:15]:
```suggestion
    const response1 = await investing('currencies/eur-usd')
    const response2 = await investing('currencies/eur-usd', 3600, 24, '1-day') // With optional params
```

File: README.md [hunk 2]
```diff
@@ -22,17 +22,20 @@ async function main () {
...
   ...
 ]
 ```
 
 
 ### Available inputs
 - [mapping.js](https://github.com/DavideViolante/investing-com-api/blob/master/mapping.js)
+- interval (in milliseconds quotes interval)
+- candle_count (maximum number of results)
+- period (n-day or n-month or n-year)
```
Comment (DavideViolante) [inline README.md:38]:
```suggestion
- input (String), see [mapping.js](https://github.com/DavideViolante/investing-com-api/blob/master/mapping.js)
- interval (Number in seconds), data interval
- candleCount (Number) Max number of results
- period (String) time window: n-day, n-month or n-year where n is a number
```

File: README.md [hunk 3]
```diff
@@ -42,3 +45,6 @@ async function main () {
 
 ### Author
 - [Davide Violante](https://github.com/DavideViolante/)
+
+### Contributor
```
Comment (DavideViolante) [inline README.md:49]:
```suggestion
### Contributors
```

File: index.js [hunk 4]
```diff
@@ -2,18 +2,18 @@ const axios = require('axios')
...
-      pair_interval: '86400', // 1 day
+      pair_interval: interval,
       chart_type: 'area', // 'area', 'candlestick'
-      candle_count: '120', // days
-      volume_series: 'yes',
-      events: 'yes',
-      period: '1-year'
+      candle_count: candleCount,
+      volume_series: 'no',
+      events: 'no',
```
Comment (DavideViolante) [inline index.js:15]: Do you know what do these params affect?

File: test/test.js [hunk 5]
```diff
@@ -18,12 +18,27 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
   })
 
-  it('should return data from investing.com', async () => {
+  it('should return data from investing.com with just input parameter', async () => {
     const response = await investing('currencies/eur-usd')
     assert.ok(response)
     assert.ok(response.length)
   })
 
+  it('shouw return data from investing.com with an 1800000 ms interval between them', async () => {
```
Comment (DavideViolante) [inline test/test.js:27]:
```suggestion
  it('shouw return data from investing.com with an 5min (1800s) interval between them', async () => {
```

File: test/test.js [hunk 6]
```diff
@@ -18,12 +18,27 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+    const response = await investing('currencies/eur-usd', 1800)
+    assert.ok((response[1].date - response[0].date) === 1800000)
```
Comment (DavideViolante) [inline test/test.js:29]:
```suggestion
    const diff = response[1].date - response[0].date
    assert.ok(diff === (1800 * 1000))
```

File: test/test.js [hunk 7]
```diff
@@ -18,12 +18,27 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 objects in a 5-hour max time window', async () => {
```
Comment (DavideViolante) [inline test/test.js:32]:
```suggestion
  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour max time window', async () => {
```

File: test/test.js [hunk 8]
```diff
@@ -18,12 +18,27 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
+    assert.ok((response[1].date - response[response.length - 1].date) <= 7200000)
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 objects in a 2 hour time window', async () => {
```
Comment (DavideViolante) [inline test/test.js:37]:
```suggestion
  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour time window', async () => {
```

File: test/test.js [hunk 9]
```diff
@@ -18,12 +18,27 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
+    assert.ok((response[1].date - response[0].date) === 1800000 && response.length === 24)
```
Comment (DavideViolante) [inline test/test.js:39]:
```suggestion
    const diff = response[1].date - response[0].date
    assert.ok(diff === (1800 * 1000))
    assert.ok(response.length === 24)
```

File: test/test.js [hunk 10]
```diff
@@ -18,12 +18,30 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
 
+  it('shouw return data from investing.com with an 5min (1800s) interval between them', async () => {
+    const response = await investing('currencies/eur-usd', 1800)
+    const diff = response[1].date - response[0].date
+    assert.ok(diff === (1800 * 1000))
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour max time window', async () => {
+    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
+    assert.ok((response[1].date - response[response.length - 1].date) <= 7200000)
```
Comment (DavideViolante) [inline test/test.js:35]:
```suggestion
    const diff = response[1].date - response[response.length - 1].date
    assert.ok(diff <= 7200000)
    assert.ok(response.length === 24)
```

File: test/test.js [hunk 11]
```diff
@@ -18,12 +18,32 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
     assert.ok(response)
     assert.ok(response.length)
   })
 
+  it('shouw return data from investing.com with an 5min (1800s) interval between them', async () => {
+    const response = await investing('currencies/eur-usd', 1800)
+    const diff = response[1].date - response[0].date
+    assert.ok(diff === (1800 * 1000))
+  })
+
```
Comment (DavideViolante) [inline test/test.js:32]:
```suggestion
  it('should return data from investing.com with an 5min (1800s) interval between them', async () => {
    const response = await investing('currencies/eur-usd', 1800)
    const diff = response[1].date - response[0].date
    assert.ok(diff === (1800 * 1000))
  })

  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 12 results', async () => {
    const response = await investing('currencies/eur-usd', 1800, 12)
    const diff = response[1].date - response[0].date
    assert.ok(diff === (1800 * 1000))
    assert.ok(response.length === 12)
  })

```

File: test/test.js [hunk 12]
```diff
@@ -18,12 +18,39 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+    assert.ok(diff <= 7200000)
+    assert.ok(response.length === 24)
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour time window', async () => {
+    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
+    const diff = response[1].date - response[0].date
+    assert.ok(diff === (1800 * 1000))
+    assert.ok(response.length === 24)
+  })
```
Comment (DavideViolante) [inline test/test.js:52]:
```suggestion
  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour time window', async () => {
    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
    const diff1 = response[1].date - response[0].date
    const diff2 = response[1].date - response[response.length - 1].date
    assert.ok(diff === (1800 * 1000))
    assert.ok(diff2 <= (7200 * 1000))
    assert.ok(response.length === 24)
  })
```

File: test/test.js [hunk 13]
```diff
@@ -18,12 +18,34 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
+    const diff = response[1].date - response[0].date
+    assert.ok(diff === (1800 * 1000))
+    assert.ok(response.length === 12)
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 24 results in a 2-hour time window', async () => {
+    const response = await investing('currencies/eur-usd', 1800, 24, '2-hour')
+    const diff1 = response[1].date - response[0].date
+    const diff2 = response[1].date - response[response.length - 1].date
+    assert.ok(diff === (1800 * 1000))
```
Comment (DavideViolante) [inline test/test.js:44]:
```suggestion
    assert.ok(diff1 === (1800 * 1000))
```

File: test/test.js [hunk 14]
```diff
@@ -18,12 +18,34 @@ describe('Tests for Investing.com unofficial APIs', () => {
...
     assert.ok(response.length)
   })
 
+  it('should return data from investing.com with an 5min (1800s) interval between them', async () => {
+    const response = await investing('currencies/eur-usd', 1800)
+    const diff = response[1].date - response[0].date
+    assert.ok(diff === (1800 * 1000))
+  })
+
+  it('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 12 results', async () => {
```
Comment (DavideViolante) [inline test/test.js:33]:
```suggestion
  xit('should return data from investing.com with an 5min (1800s) interval between them and a maximum of 12 results', async () => {
```

File: index.js (hunk 4 above)
Comment (gustavomfc) [reply index.js:15]: Yep! When "events" is setted to 'yes' the response comes with a object with some links to news related to the index queried, "volume_series" when setted to 'yes' also bring some extra content to reply too, but useless to the final object, I've setted to 'no' to make response processing lighter.
